detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerDetector.ts

```diff
@@ -74,4 +94,4 @@ export class DockerCGroupV1Detector implements Detector {
   }
 }
 
-export const dockerCGroupV1Detector = new DockerCGroupV1Detector();
+export const dockerDetector = new DockerDetector();
```

Comment:
is this name change a breaking change for existing users?

Reply:
It is - if they update - they will have to change the detector name - but this name is scalable and would not need a future change.

I shall update the Read.me

Reply:
why not ContainerDetector if it is compatible with containerd as well?

Reply:
@svrnm makes sense - I shall rename it to that.

Reply:
The question is, do we want end-users to swap out that detector every time they move from one container engine to another? Maybe they are not even aware that things got switched out because that's not the developers decision.

The detection provided here works for at least 2 engines we have tests, we could try out podman (and others) as well.

Actually it would be great to have this fixed via OCI spec (see this issue: https://github.com/opencontainers/runtime-spec/issues/1105), but that's a long standing issue.

Reply:
> The question is, do we want end-users to swap out that detector every time they move from one container engine to another? Maybe they are not even aware that things got switched out because that's not the developers decision.
> 
> The detection provided here works for at least 2 engines we have tests, we could try out podman (and others) as well.
> 
> Actually it would be great to have this fixed via OCI spec (see this issue: [opencontainers/runtime-spec#1105](https://github.com/opencontainers/runtime-spec/issues/1105)), but that's a long standing issue.

I think both options are valid and have pros and cons.
Since it's already implemented for general container, I support keeping it as it is now :)

Thank you for the reference to the OCI spec